src/sbn_probability.cpp

```diff
@@ -179,3 +239,32 @@ EigenVectorXd SBNProbability::ProbabilityOf(
   }
   return results;
 }
+
+double SBNProbability::ProbabilityOfLogSpace(
```

Comment:
It seems to me that these functions should replace `ProbabilityOf` rather than be alongside. We are always going to be keeping `sbn_parameters_` in log space moving forward. Also, how they are kept doesn't really matter to the user of this file.

Reply:
Agree. But these functions are called by the user after doing EM training as well, which does not keep log values in sbn_parameters_ yet. One possibility until EM testing is done, is to write one line at the end of EM training to transform sbn_parameters_ to log. Then, we can replace `ProbabilityOf` 
 by `ProbabilityOfInLog`.

Reply:
Yes, I like your suggestion!